src/worker.cpp

```diff
@@ -651,12 +652,65 @@ bool Worker::isCPUIOGoodOnly(const std::string& pgKeyword)
     return false;
 }
 
-void Worker::primeInventory(const types::IPZVpdMap& ipzVpdMap,
-                            types::ObjectMap primeObjects)
+bool Worker::primeInventory(const std::string& i_vpdFilePath)
 {
-    // TODO implementation
-    (void)ipzVpdMap;
-    (void)primeObjects;
+    logging::logMessage("primeInventory called for FRU " + i_vpdFilePath);
+
+    if (i_vpdFilePath.empty())
+    {
+        logging::logMessage("Empty VPD file path given");
+        return false;
+    }
+
+    if (m_parsedJson.empty())
+    {
+        logging::logMessage("Empty JSON detected for " + i_vpdFilePath);
+        return false;
+    }
+
+    types::ObjectMap l_objectInterfaceMap;
+    for (const auto& l_Fru : m_parsedJson["frus"][i_vpdFilePath])
+    {
+        types::InterfaceMap l_interfaces;
+        sdbusplus::message::object_path l_fruObjectPath(l_Fru["inventoryPath"]);
+
+        types::VPDMapVariant l_emptyValue;
+
+        // Add extra interfaces mentioned in the Json config file
+        if (l_Fru.contains("extraInterfaces"))
+        {
+            populateInterfaces(l_Fru["extraInterfaces"], l_interfaces,
+                               l_emptyValue);
+        }
+
+        types::PropertyMap presProp;
```

Comment:
What is "presProp". Please use meaningful names/abbreviations.

Reply:
updated.